src/merkle_tree/mod.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{hash, hash_pair};
+
+pub struct MerkleTree {
+    hashes: Vec<Vec<u8>>,
+    count: usize
+}
+
+impl MerkleTree {
+    pub fn new(elements: Vec<String>) -> MerkleTree{
+        let leaves = Self::hash_elements(elements.clone());
+        let hashes = Self::build_hashes(leaves);
+        MerkleTree{hashes, count: elements.len()}
+    }
+
+    pub fn root_hash(tree: MerkleTree) -> Vec<u8> {
+        tree.hashes[0].clone()
+    }
+
+    pub fn add(tree: MerkleTree, elements: Vec<String>) -> MerkleTree {
+        let new_leaves = Self::hash_elements(elements);
+        let old_leaves = tree.hashes[{tree.count-1}..].to_vec();
+        let leaves = [old_leaves, new_leaves].concat();
+        let hashes = Self::build_hashes(leaves.clone());
+        MerkleTree{hashes, count: leaves.len()}
+    }
+
+    fn hash_elements(elements: Vec<String>) -> Vec<Vec<u8>> {
```

Comment:
```suggestion
    fn hash_elements(&self, elements: &[String]) -> Vec<Vec<u8>> {
```
same as above, you can pass `self` as argument to make it a method of the struct. also it is better to pass `&[String]` than `Vec` because thus the functions may accept also arrays and not just vectors.